src/main/scala/tilelink/SRAM.scala

```diff
@@ -201,8 +207,8 @@ class TLRAM(
       if (code.canDetect) code.encode(d, p) else code.encode(d)
     })
 
-    d_raw_data := mem.read(addr, ren)
-    when (wen) { mem.write(addr, coded, sel) }
+    d_raw_data := mem.mem.read(addr, ren)
```

Comment:
This looks a little bit awkward and confusing now, with the double `mem`. Could we give the outer `mem` a different name up on line 61 here?

This kind of change will need to be replicated to all the other SRAM instances for the other protocols.